lib/controllers/share.js

```diff
@@ -83,7 +83,8 @@ function S3(serviceOptions) {
     } else {
         var aws = require('aws-sdk');
         aws.config.update({
-            region: serviceOptions.region
+          endpoint: serviceOptions.endpoint ?? undefined,
```

Comment:
The surrounding code seems to use 4 spaces for indentation size, so these two lines should be indented 2 spaces more.